### PR TITLE
purge lazy

### PR DESCRIPTION
**Pull request by wence-**
Remove all traces of lazy evaluation.

Also trims some fat from the strong scaling limits.

### CODE REVIEW

**Comment by wence-** [comment]
Needs to land simultaneously with firedrakeproject/firedrake#1490

File: pyop2/configuration.py [hunk 1]
```diff
@@ -61,10 +60,8 @@ class Configuration(dict):
          to a node-local filesystem too.
     :param log_level: How chatty should PyOP2 be?  Valid values
         are "DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL".
-    :param lazy_evaluation: Should lazy evaluation be on or off?
-    :param lazy_max_trace_length: How many :func:`par_loop`\s
-        should be queued lazily before forcing evaluation?  Pass
-        `0` for an unbounded length.
+    :param dump_gencode: Should PyOP2 write the generated code
+]        somewhere for inspection?
```
Comment (miklos1) [inline pyop2/configuration.py]: Right bracket?

File: test/unit/test_extrusion.py [hunk 2]
```diff
@@ -37,7 +37,40 @@
...
+                    for j in range(0, mesh2d[i]):
+                        m = map[mm][c]
+                        for k in range(0, len2):
+                            ind[count] = m*(layers - d) + A[d][k] + offset
+                            count += 1
+                        c += 1
+                elif dofs[i, 1-d] != 0:
+                    c += mesh2d[i]
+                offset += a4*nums[i]*(layers - d)
+    return ind
```
Comment (miklos1) [inline test/unit/test_extrusion.py]: Transcription from `computeind.pyx`.  No need for Cython when it's only used in a test.

File: pyop2/base.py [hunk 3]
```diff
@@ -825,7 +653,9 @@ def _wrapper_cache_key_(self):
 
     def __getattr__(self, name):
         """Returns a :class:`Set` specific attribute."""
-        return getattr(self._parent, name)
+        value = getattr(self._parent, name)
+        setattr(self, name, value)
+        return value
```
[the remark this answers is not in the document]
Comment (wence-) [reply pyop2/base.py]: Correct.

**Pull request merged**